src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,193 @@
+   */
+  onDrawerMenuStateChange: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of an extensions item.
+   * Ex: `onSelectExtensionsItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectExtensionItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Help utility item.
+   * If `onSelectHelp` is not provided, the Help utility item will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectHelp: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Logout action button.
+   * If `onSelectLogout` is not provided, the Logout action button will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectLogout: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a navigation item.
+   * Ex: `onSelectNavigationItem(String selectedNavigationItemKey, Object metaData)`
+   */
+  onSelectNavigationItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Settings utility item.
+   * If `onSelectSettings` is not provided, the Settings utility item will not be rendered.
+   * Ex: `onSelectSettings()`
+   */
+  onSelectSettings: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a custom utility item.
+   * Ex: `onSelectUtilityItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectUtilityItem: PropTypes.func,
+  /**
+   * A configuration object that defines the strings rendered within the ApplicationNavigation header.
+   */
+  titleConfig: titleConfigPropType,
+  /**
+   * A configuration object with information pertaining to the application's user.
+   */
+  userConfig: userConfigPropType,
+  /**
+   * An array of configuration objects with information specifying the creation of additional utility menu items.
+   * These items are rendered within the popup utility menu at larger breakpoints and within the drawer menu at smaller breakpoints.
+   */
+  utilityItems: utilityItemsPropType,
+};
+
+const ApplicationNavigation = ({
+  activeNavigationItemKey,
+  children,
+  disablePromptsForLogout,
+  disablePromptsForNavigationItems,
+  extensionItems,
+  hero,
+  navigationItems,
+  navigationPromptResolutionOptions,
+  notifications,
+  onDrawerMenuStateChange,
+  onSelectExtensionItem,
+  onSelectHelp,
+  onSelectLogout: propOnSelectLogout,
+  onSelectNavigationItem: propOnSelectNavigationItem,
+  onSelectSettings,
+  onSelectUtilityItem,
+  titleConfig,
+  userConfig,
+  utilityItems,
+}) => {
+  const navigationPromptCheckpointRef = useRef();
+
+  const onSelectNavigationItem = useCallback((selectedItemKey) => {
+    if (disablePromptsForNavigationItems) {
+      propOnSelectNavigationItem(selectedItemKey);
+      return;
+    }
+
+    navigationPromptCheckpointRef.current.resolvePrompts(navigationPromptResolutionOptions).then(() => {
+      propOnSelectNavigationItem(selectedItemKey);
+    }).catch((e) => { if (e) throw e; });
```

Comment:
Why catch if we are just going to re-throw it?

Reply:
The NavigationPromptCheckpoint rejects its returned Promise when the user opts to reject the prompt, but it doesn't return an error with the rejection. This catch will swallow that error-free rejection and but re-throw any present errors that originated from something else.

Here I'm just trying to keep the 'Uncaught error in Promise' logs from showing up in the console when users reject the prompt.